Customizing/global/plugins/Services/Cron/CronHook/ReportMaster/classes/ReportBase/class.catFilter.php

```diff
@@ -1203,4 +1203,107 @@ public function setPreSelect(array $pre_select) {
 	public function uncheckRecursiveSearch() {
 		$this->search_recursive = false;
 	}
+}
+
+/**
+ *	This class introduces a consistent course type filter.
+ *	To ensure performance this filter will work by an indirect priciple:
+ *	Instead of operating on topics in the report query it will constrain
+ *	the range of valid crs_ids after checking for relevant courses.
+ */
+class courseTopicsFilter {
+	protected $field;
+	protected $recursive;
+	protected $filter;
+	protected $crs_ids_row;
+	protected $id;
+	protected $gIldb;
+
+	public function __construct($id, $crs_ids_row) {
+		$this->id = $id;
+		$this->crs_ids_row = $crs_ids_row;
+		global $ilDB,$lng;
+		$this->gIldb = $ilDB;
+		$this->lng = $lng;
```

Comment:
$this->gIlDB => $this->gDB
$this->lng => $this->gLng

So war das eigentlich mal abgemacht.